x/model/types/messages_model_version_test.go

```diff
@@ -1,74 +1,736 @@
+				msg.Vid = 65535
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "Pid == 1",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.Pid = 1
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "Pid == 65535",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.Pid = 65535
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "OtaUrl is omitted",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.OtaUrl = ""
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "OtaUrl length == 256",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.OtaUrl = "https://sampleflowurl.dclauth/" + tmrand.Str(256-30) // length = 256
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "MinApplicableSoftwareVersion and MaxApplicableSoftwareVersion are not set",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.MinApplicableSoftwareVersion = 0
+				msg.MaxApplicableSoftwareVersion = 0
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "MinApplicableSoftwareVersion is set and MaxApplicableSoftwareVersion is not set",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.MinApplicableSoftwareVersion = 1
+				msg.MaxApplicableSoftwareVersion = 0
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "MinApplicableSoftwareVersion is not set and MaxApplicableSoftwareVersion is set",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.MinApplicableSoftwareVersion = 0
+				msg.MaxApplicableSoftwareVersion = 1
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "MinApplicableSoftwareVersion and MaxApplicableSoftwareVersion are set " +
+				"and MinApplicableSoftwareVersion < MaxApplicableSoftwareVersion",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.MinApplicableSoftwareVersion = 5
+				msg.MaxApplicableSoftwareVersion = 10
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "MinApplicableSoftwareVersion and MaxApplicableSoftwareVersion are set " +
+				"and MinApplicableSoftwareVersion == MaxApplicableSoftwareVersion",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.MinApplicableSoftwareVersion = 7
+				msg.MaxApplicableSoftwareVersion = 7
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "ReleaseNotesUrl is omitted",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.ReleaseNotesUrl = ""
+				return msg
+			}(validMsgUpdateModelVersion()),
+		},
+		{
+			name: "ReleaseNotesUrl length == 256",
+			msg: func(msg *MsgUpdateModelVersion) *MsgUpdateModelVersion {
+				msg.ReleaseNotesUrl = "https://sampleflowurl.dclauth/" + tmrand.Str(256-30) // length = 256
+				return msg
+			}(validMsgUpdateModelVersion()),
```

Comment:
Please add the following positive case:
- SoftwareVersion=0

Reply:
Done.